packages/base/card-api.gts

```diff
@@ -1597,14 +1597,18 @@ export function isCard(card: any): card is CardDef {
   return card && typeof card === 'object' && isBaseInstance in card;
 }
 
+export function isCompoundField(maybeField: any): maybeField is FieldDef {
+  return maybeField?.constructor?.isFieldDef && isBaseInstance in maybeField;
```

Comment:
isn't a compound field any fieldDef that doesn't have a `[primitive]` symbol? probably we should include that in this type assertion?

Reply:
conversely you could have a function called `isPrimitiveField()` that returns if the field is a `FieldDef` has has the `[primitive]` symbol. these functions should be mutually exclusive for any `FieldDef`

Reply:
renaming `isCard()` to `isCardOrField()` and adding an actual `isCard()` removed the need to create isPrimitiveField or isCompoundField